core/src/test/scala/unit/kafka/server/DynamicConfigChangeTest.scala

```diff
@@ -539,4 +545,52 @@ class DynamicConfigChangeUnitTest {
     //Then
     assertEquals(Seq(), result)
   }
+
+  @Test
+  def testEnableRemoteLogStorageOnTopic(): Unit = {
+    val topic = "test-topic"
+    val tp = new TopicPartition(topic, 0)
+    val partition: Partition = mock(classOf[Partition])
+    when(partition.isLeader).thenReturn(true)
+    val rlm: RemoteLogManager = mock(classOf[RemoteLogManager])
+    val leaderPartitionsArg: ArgumentCaptor[util.Set[Partition]] = ArgumentCaptor.forClass(classOf[util.Set[Partition]])
+    val followerPartitionsArg: ArgumentCaptor[util.Set[Partition]] = ArgumentCaptor.forClass(classOf[util.Set[Partition]])
+    doNothing().when(rlm).onLeadershipChange(leaderPartitionsArg.capture(), followerPartitionsArg.capture(), any())
+    val replicaManager: ReplicaManager = mock(classOf[ReplicaManager])
+    when(replicaManager.remoteLogManager).thenReturn(Some(rlm))
+    when(replicaManager.onlinePartition(tp)).thenReturn(Some(partition))
+    val log: UnifiedLog = mock(classOf[UnifiedLog])
+    when(log.remoteLogEnabled()).thenReturn(true)
+    when(log.topicPartition).thenReturn(tp)
+
+    val isRemoteLogEnabledBeforeUpdate = false
+    val configHandler: TopicConfigHandler = new TopicConfigHandler(replicaManager, null, null, null)
+    configHandler.maybeBootstrapRemoteLogComponents(Seq(log), isRemoteLogEnabledBeforeUpdate)
+    assertTrue(followerPartitionsArg.getValue.isEmpty)
+    assertEquals(Collections.singleton(partition), leaderPartitionsArg.getValue)
+  }
+
+  @Test
+  def testDisableRemoteLogStorageOnTopic(): Unit = {
+    val topic = "test-topic"
+    val tp = new TopicPartition(topic, 0)
+    val partition: Partition = mock(classOf[Partition])
+    when(partition.isLeader).thenReturn(true)
+    when(partition.isLeader).thenReturn(false)
```

Comment:
I was under the impression that this line will completely override the previous one, can you whether the previous line does something?

Reply:
If you want the first invocation to return true and the second to return false I think the correct syntax is:
```
when(partition.isLeader).thenReturn(true).thenReturn(false)
```